device/dell/x86_64-dellemc_z9432f_c3758-r0/plugins/psuutil.py

```diff
@@ -39,84 +40,89 @@ def isDockerEnv(self):
+            ret_status, ipmi_cmd_ret = getstatusoutput_noshell(self.IPMI_PSU_POUT)
+            if ret_status == 0:
+                rdata = ipmi_cmd_ret.splitlines()[3].split(':')[1].split(' ')[1]
+                return rdata
 
         except Exception:
-           logging.error('Failed to execute : %s'%self.IPMI_PSU_POUT.format(index))
+            logging.error('Failed to execute : %s'%(' '.join(self.IPMI_PSU_POUT)))
 
     def get_psu_fan_speed(self,index):
         try:
-           ret_status, ipmi_cmd_ret = commands.getstatusoutput(self.IPMI_PSU_FAN_SPEED.format(index))
-           if ret_status == 0:
-              rdata = ipmi_cmd_ret.splitlines()[3].split(':')[1].split(' ')[1]
-              return rdata
+            self.IPMI_PSU_FAN_SPEED[3] = 'PSU' + str(index) + '_Fan'
+            ret_status, ipmi_cmd_ret = getstatusoutput_noshell(self.IPMI_PSU_FAN_SPEED)
+            if ret_status == 0:
+                rdata = ipmi_cmd_ret.splitlines()[3].split(':')[1].split(' ')[1]
+                return rdata
 
         except Exception:
-           logging.error('Failed to execute : %s'%self.IPMI_PSU_FAN_SPEED.format(index))
+            logging.error('Failed to execute : %s'%(' '.join(self.IPMI_PSU_FAN_SPEED)))
 
 
     #Fetch FRU Data for given fruid
     def get_psu_airflow(self, index):
         if index == 1:
-           fru_id = 'FRU_PSUL'
+            fru_id = 'FRU_PSUL'
         else:
-           fru_id = 'FRU_PSUR'
+            fru_id = 'FRU_PSUR'
 
-        ret_status, ipmi_cmd_ret = commands.getstatusoutput(self.IPMI_FRU)
+        ret_status, ipmi_cmd_ret = getstatusoutput_noshell(self.IPMI_FRU)
         if ret_status:
-           logging.error('Failed to execute ipmitool: '+ self.IPMI_FRU)
+            logging.error('Failed to execute ipmitool: '+ self.IPMI_FRU)
 
         found_fru = False
         for line in ipmi_cmd_ret.splitlines():
             if line.startswith('FRU Device Description') and fru_id in line.split(':')[1] :
-               found_fru = True
+                found_fru = True
             if found_fru and line.startswith(' Board Product '):
-               return ' B2F' if 'PS/IO' in line else ' F2B'
+                return ' B2F' if 'PS/IO' in line else ' F2B'
         return ''
 
     # Read FRU info
     def get_fru_info(self,fru_id,reg_name):
         output = None
         Found = False
         try:
-           status, ipmi_fru_list = commands.getstatusoutput(self.IPMI_FRU_DATA.format(fru_id))
-           if status == 0:
-              for item in ipmi_fru_list.split("\n"):
-                  if reg_name == item.split(':')[0].strip(' '):
-                     output = item.strip()
-                     output = output.split(':')[1]
-                     Found = True
-                     break;
-
-              if not Found:
-                 logging.error('\nFailed to fetch: ' +  reg_name + ' sensor ')
-
-              return output
+            self.IPMI_FRU_DATA[3] += str(fru_id)
```

Comment:
>self.IPMI_FRU_DATA[3] [](http://example.com/codeflow?start=12&length=21)

This line will change the member var, so later function call to `get_fru_info()` will be impacted. It is a bug.